shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateIndexStatementValidator.java

```diff
@@ -38,7 +39,9 @@ public void preValidate(final ShardingRule shardingRule, final SQLStatementConte
                             final List<Object> parameters, final ShardingSphereSchema schema) {
         validateTableExist(schema, Collections.singletonList(sqlStatementContext.getSqlStatement().getTable()));
         String tableName = sqlStatementContext.getSqlStatement().getTable().getTableName().getIdentifier().getValue();
-        String indexName = sqlStatementContext.getSqlStatement().getIndex().getIdentifier().getValue();
+        String indexName = null != sqlStatementContext.getSqlStatement().getIndex() 
+                ? sqlStatementContext.getSqlStatement().getIndex().getIdentifier().getValue()
+                : IndexMetaDataUtil.getGeneratedLogicIndexName(sqlStatementContext.getSqlStatement().getColumns());
```

Comment:
How about `sqlStatementContext.getIndexName()`?

Reply:
> How about `sqlStatementContext.getIndexName()`?

@tristaZero Great idea, I have extracted the logic into the getIndexes method in sqlStatementContext.